cmd/release-controller/cluster_distribution.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+type ClusterDistribution interface {
+	Get() string
+	Contains(str string) bool
+}
+
+var ErrEmptyItems = errors.New("cannot create cluster distribution with no items")
+
+type RandomClusterDistribution struct {
+	pool    []string
+}
+
+func NewRandomClusterDistribution(items ...string) (ClusterDistribution, error) {
+	if len(items) == 0 {
+		return nil, ErrEmptyItems
+	}
+	rand.Seed(time.Now().UnixNano())
```

Comment:
`rand.New(rand.NewSource(time.Now().UnixNano())` and store the result on your instance I think

Reply:
Done!